lib/srv/desktop/tdp/conn_test.go

```diff
@@ -46,7 +46,7 @@ func TestTDPConnTracksLocalRemoteAddrs(t *testing.T) {
 		},
 		{
 			desc:   "does not implement srv.TrackingConn",
-			conn:   &bytes.Buffer{},
+			conn:   &fakeConn{Buffer: &bytes.Buffer{}},
```

Comment:
Consider `io.NopCloser` to add the close method instead of implementing your own.

Reply:
`NopCloser` returns a `ReadCloser` but I need a `ReadWriteCloser` here